addons/adf/src/main/java/org/apache/falcon/adfservice/ADFJob.java

```diff
@@ -0,0 +1,563 @@
+    }
+
+    protected String getScriptPath() throws FalconException {
+        if (!activityHasScriptPath()) {
+            throw new FalconException("JSON object does not have object: "
+                    + ADFJsonConstants.ADF_REQUEST_SCRIPT_PATH);
+        }
+
+        try {
+            String scriptPath = activityExtendedProperties.getString(ADFJsonConstants.ADF_REQUEST_SCRIPT_PATH);
+            if (StringUtils.isBlank(scriptPath)) {
+                throw new FalconException("JSON object " + ADFJsonConstants.ADF_REQUEST_SCRIPT_PATH + " not"
+                        + " found or empty in ADF request.");
+            }
+            return scriptPath;
+        } catch (JSONException jsonException) {
+            throw new FalconException("Error when parsing ADF JSON object: "
+                    + activityExtendedProperties, jsonException);
+        }
+    }
+
+    protected String getScriptContent() throws FalconException {
+        if (activityHasScriptPath()) {
+            throw new FalconException("JSON object does not have object: " + ADFJsonConstants.ADF_REQUEST_SCRIPT);
+        }
+        try {
+            String script = activityExtendedProperties.getString(ADFJsonConstants.ADF_REQUEST_SCRIPT);
+            if (StringUtils.isBlank(script)) {
+                throw new FalconException("JSON object " + ADFJsonConstants.ADF_REQUEST_SCRIPT + " cannot"
+                        + " be empty in ADF request.");
+            }
+            return script;
+        } catch (JSONException jsonException) {
+            throw new FalconException("Error when parsing ADF JSON object: "
+                    + activityExtendedProperties, jsonException);
+        }
+    }
+
+    protected String getClusterNameToRunProcessOn() throws FalconException {
+        return getClusterName(getHadoopLinkedService());
+    }
+
+    protected Entity submitAndScheduleJob(String entityType, String msg) throws FalconException {
+        Entity entity = jobManager.submitJob(entityType, msg);
+        jobManager.scheduleJob(entityType, jobEntityName());
+        return entity;
+    }
+
+    private String getHadoopLinkedService() throws FalconException {
+        String hadoopLinkedService;
+        try {
+            hadoopLinkedService = activity.getString(ADFJsonConstants.ADF_REQUEST_LINKED_SERVICE_NAME);
+        } catch (JSONException jsonException) {
+            throw new FalconException("Error when parsing ADF JSON object: "
+                    + activity, jsonException);
+        }
+
+        if (StringUtils.isBlank(hadoopLinkedService)) {
+            throw new FalconException("JSON object " + ADFJsonConstants.ADF_REQUEST_LINKED_SERVICE_NAME
+                    + " in the activity cannot be empty in ADF request.");
+        }
+        return hadoopLinkedService;
+    }
+
+    protected void startProcess(Feed inputFeed, Feed outputFeed,
+                                String engineType, String scriptPath) throws FalconException {
```

Comment:
engineType provides process type (either hive or pig for now). I added it to the log.